### PR TITLE
support consumption plan as default

### PR DESCRIPTION
**Pull request by yiliuTo**
This is pr is to make the AZD template support both ASA consumption plan and standard plan, and the default option will be the consumption plan.

To test it, you can run:
```
mkdir ASA-Samples-Event-Driven-Application
cd ASA-Samples-Event-Driven-Application
azd auth login
azd config set alpha.springapp on
azd init --template https://github.com/yiliuTo/ASA-Samples-Event-Driven-Application/ -b consumption-default
azd up
```

The template by default uses consumption plan, and if users want to use standard, they can set `azd env set PLAN standard` then run `azd up`. If users have already provisioned the resources with the consumption plan and want to try the Standard plan, they need to run `azd down` first to delete the resources, and then run `azd env set PLAN standard` and `azd up` again to provision and deploy.

### CODE REVIEW

**Comment by yiliuTo** [comment]
We will not merge this pr until the corresponding MS doc gets merged. cc @moarychan and @Netyyyy

**Comment by moarychan** [comment]
The MS doc is live, please help to approve and merge this PR, thanks!

File: README.md
Comment (fangjian0423) [inline README.md]: Confuse for the plan name, below which one is correct?

- consumption plan
- consumption and dedicated plan
- Standard consumption and dedicated plan

File: README.md
Comment (fangjian0423) [inline README.md]: Using `Azure Sping Apps` in REAMDE looks better than ASA.

File: README.md
Comment (yiliuTo) [reply README.md]: will use `Standard consumption and dedicated plan`

**Review by fangjian0423 — APPROVED** [review]
LGTM, thanks @yiliuTo

**Pull request merged**
